x-pack/legacy/plugins/lens/server/usage/collectors.ts

```diff
@@ -21,25 +21,124 @@ export function registerLensUsageCollector(
       };
     };
     config: KibanaConfig;
+    server: Server;
   }
 ) {
+  let isCollectorReady = false;
+  async function determineIfTaskManagerIsReady() {
+    let isReady = false;
+    try {
+      isReady = await isTaskManagerReady(plugins.server);
+    } catch (err) {} // eslint-disable-line
+
+    if (isReady) {
+      isCollectorReady = true;
+    } else {
+      setTimeout(determineIfTaskManagerIsReady, 500);
+    }
+  }
+  determineIfTaskManagerIsReady();
+
   const lensUsageCollector = plugins.usage.collectorSet.makeUsageCollector({
     type: 'lens',
-    fetch: async (callCluster: CallCluster): Promise<LensUsage> => {
+    fetch: async (): Promise<LensUsage> => {
       try {
-        return getVisualizationCounts(callCluster, plugins.config);
+        const docs = await getLatestTaskState(plugins.server);
+        // get the accumulated state from the recurring task
+        const state: LensTelemetryState = get(docs, '[0].state');
+
+        const dates = Object.keys(state.byDate || {}).map(dateStr => parseInt(dateStr, 10));
+        const suggestionDates = Object.keys(state.suggestionsByDate || {}).map(dateStr =>
+          parseInt(dateStr, 10)
+        );
+
+        const eventsLast30: Record<string, number> = {};
+        const eventsLast90: Record<string, number> = {};
+        const suggestionsLast30: Record<string, number> = {};
+        const suggestionsLast90: Record<string, number> = {};
+
+        const last30 = moment()
+          .subtract(30, 'days')
+          .unix();
+        const last90 = moment()
+          .subtract(90, 'days')
+          .unix();
+
+        dates.forEach(date => {
+          if (date >= last30) {
+            addEvents(eventsLast30, state.byDate[date]);
+            addEvents(eventsLast90, state.byDate[date]);
+          } else if (date > last90) {
+            addEvents(eventsLast90, state.byDate[date]);
+          }
+        });
+
+        suggestionDates.forEach(date => {
+          if (date >= last30) {
+            addEvents(suggestionsLast30, state.suggestionsByDate[date]);
+            addEvents(suggestionsLast90, state.suggestionsByDate[date]);
+          } else if (date > last90) {
+            addEvents(suggestionsLast90, state.suggestionsByDate[date]);
+          }
+        });
+
+        return {
+          ...state.saved,
+          events_30_days: eventsLast30,
+          events_90_days: eventsLast90,
+          suggestion_events_30_days: suggestionsLast30,
+          suggestion_events_90_days: suggestionsLast90,
+        };
       } catch (err) {
         return {
-          saved_total: 0,
-          saved_last_30_days: 0,
-          saved_last_90_days: 0,
-          visualization_types_overall: {},
-          visualization_types_last_30_days: {},
-          visualization_types_last_90_days: {},
+          saved_overall_total: 0,
```

Comment:
I wonder if we should add some error field to the object, just so we can filter out any that are written due to an error state.